### PR TITLE
Skip tests that require portable assets on community architectures.

### PR DESCRIPTION
**Pull request by tmds**
For community architectures, only non-portable assets are available.

@ViktorHofer @MichaelSimons ptal.

cc @omajid @Swapnali911

### CODE REVIEW

**Review by MichaelSimons — APPROVED** [review]

File: src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTest.cs [hunk 1]
```diff
@@ -29,6 +29,14 @@ internal void Execute(DotNetSdkHelper dotNetHelper, string testRoot, string[]? f
...
         string customNewArgs = Template.IsAspNetCore() && NoHttps ? "--no-https" : string.Empty;
         string projectDirectory = Path.Combine(testRoot, projectName);
+        string portableRid = $"linux-{TargetArchitecture}";
+
+        // We don't need to separately execute with the portable rid when it is the same as the TargetRid.
+        // When they are different (that is: the build is non-portable), the portable assets needed for the tests were produced by a different build.
+        // These portable assets are not available for community supported architectures.
+        bool executeWithPortableRid = TargetRid != portableRid
+                                      && TargetArchitecture != "s390x"
+                                      && TargetArchitecture != "ppc64le";
```
Comment (omajid) [inline src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTest.cs:39]: Should we proactively add all non-Microsoft architectures from https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.architecture?view=net-8.0#fields to here?

Comment (tmds) [reply src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTest.cs:39]: I've included the ones that are causing test failures for us. Can you list what additional ones I should add?

Comment (omajid) [reply src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTest.cs:39]: Looking at https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.architecture?view=net-9.0#fields : 

- `loongarch64`
- `riscv64`

**Comment by tmds** [comment]
I've refactored this to use a Trait (similar to https://github.com/dotnet/scenario-tests/pull/113).

File: src/Microsoft.DotNet.ScenarioTests.Common/Program.cs [hunk 2]
```diff
@@ -249,21 +249,30 @@ private static List<string> CreateBuildTraits(string dotnetRoot, string targetRi
         {
             List<string> buildTraits = new();
 
+            int archSeparatorPos = targetRid.LastIndexOf('-');
+            string ridWithoutArch = targetRid.Substring(0, archSeparatorPos != -1 ? archSeparatorPos : 0);
+            string arch = targetRid.Substring(archSeparatorPos + 1);
```
Comment (mthalman) [inline src/Microsoft.DotNet.ScenarioTests.Common/Program.cs:254]:
```suggestion
            string[] ridParts = targetRid.Split('-');
            string ridWithoutArch = ridParts[0];
            string arch = ridParts[1];
```

Comment (tmds) [reply src/Microsoft.DotNet.ScenarioTests.Common/Program.cs:254]: This doesn't take arch as the last part (like in `linux-musl-x64`).

**Review by mthalman — APPROVED** [review]

**Comment by tmds** [comment]
@mthalman will this still go into the vmr 9.0?

**Pull request merged**